packages/ses/src/error/note-log-args.js

```diff
@@ -118,33 +127,42 @@ export const makeLRUCacheMap = budget => {
     return cell;
   };
 
+  /**
+   * @param {K} key
+   */
   const has = key => touchCell(key) !== undefined;
   freeze(has);
 
+  /**
+   * @param {K} key
+   */
   // TODO Change to the following line, once our tools don't choke on `?.`.
   // See https://github.com/endojs/endo/issues/1514
-  // const get = key => touchCell(key)?.value;
+  // const get = key => touchCell(key)?.keyValue.get(key);
   const get = key => {
     const cell = touchCell(key);
-    return cell && cell.value;
+    return cell && cell.keyValue.get(key);
   };
   freeze(get);
 
+  /**
+   * @param {K} key
+   * @param {V} value
+   */
   const set = (key, value) => {
     if (budget >= 1) {
       let cell = touchCell(key);
       if (cell !== undefined) {
-        cell.value = value;
+        cell.keyValue.set(key, value);
       } else {
         if (size >= budget) {
           const condemned = head.prev;
           spliceOut(condemned); // Drop least recently used
```

Comment:
We need to also null out the `keyValue` property of the condemned cell. If the key is kept around, so will the cell through `keyToCell`, even though the cell is unlinked from others.

It might then make sense to let `set` create this `WeakMap` and set it on the cell on insertion below (and keep `makeSelfCell` agnostic of the data kept in the cell.